notifier/notifier.go

```diff
@@ -467,7 +462,35 @@ func (n *Manager) sendAll(alerts ...*Alert) bool {
 		numSuccess uint64
 	)
 	for _, ams := range amSets {
+		var (
+			b   []byte
+			err error
+		)
+
 		ams.mtx.RLock()
+		defer ams.mtx.RUnlock()
+
+		switch ams.cfg.APIVersion {
```

Comment:
Maybe it could be cached to avoid serializing several times the same data.

Reply:
Good point. Done.